cupyx/scipy/sparse/compressed.py

```diff
@@ -175,10 +176,15 @@ class _compressed_sparse_matrix(sparse_data._data_matrix,
         }
         ''', 'min_nonzero_reduction')
 
-    _max_arg_reduction_kern = core.RawKernel(r'''
-        extern "C" __global__
-        void max_arg_reduction(double* data, int* indices, int* x, int* y,
-                               int length, long long* z) {
+    # For _max_arg_reduction_mod and _min_arg_reduction_mod below, we pick
+    # the right template specialization according to input dtypes at runtime.
+    # The distinction in int types (T2) is important for portability in OS.
+
+    _max_arg_reduction_mod = core.RawModule(
+        code=r'''
+        template<typename T1, typename T2> __global__ void
+        max_arg_reduction(T1* data, int* indices, int* x, int* y,
```

Comment:
Optional: these two kernels are essentially the same but with just a single operator changing.
Can we templatize them? even just a single python f-string template is fine :).

You can ignore this, or do it in another PR if you want!

Thanks

Reply:
Yeah I wanted to do this too but for some reason I forgot. Done in 82a6dc0.